### PR TITLE
Score

### PR DESCRIPTION
**Pull request by michbr**

### CODE REVIEW

File: Assets/Scripts/GameControl/GameModes/Score.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+public class Score : MonoBehaviour {
+
+	public float value;
+	public ClientController owner;
+	public bool teamOwned;
+
+	void Start() {
+		Label label = GetComponent<Label>();
+		if (label != null) {
```
Comment (semimono) [inline Assets/Scripts/GameControl/GameModes/Score.cs]: Should we print a warning otherwise?

File: Assets/Scripts/GenericRifle.cs [hunk 2]
```diff
@@ -99,11 +99,12 @@ public class GenericRifle : NetworkBehaviour {
         RaycastHit hitInfo;
         bool hit = Physics.Raycast(position, direction, out hitInfo, range);
         if (hit) {
-            Health health = getParentComponent<Health>(hitInfo.collider.transform);
-            if (health != null) {
-                bulletHitHealth(direction, hitInfo.point, hitInfo.normal, health.netId);
+	        Label label = getParentComponent<Label>(hitInfo.collider.transform);
+            //Health health = getParentComponent<Health>(hitInfo.collider.transform);
```
Comment (semimono) [inline Assets/Scripts/GenericRifle.cs]: Could we just remove this?

File: Assets/Scripts/Health.cs [hunk 3]
```diff
@@ -47,12 +51,12 @@ public class Health : NetworkBehaviour {
...
 	public virtual void destruct() {
-		Label label = GetComponent<Label>();
 		if (label != null)
-			label.sendTrigger(gameObject, new DestructTrigger());
+			label.sendTrigger(lastAttacker, new DestructTrigger());
 	}
 
-	public virtual void hurt(float pain) {
+	public virtual void hurt(float pain, GameObject instigator) {
+		lastAttacker = instigator;
```
Comment (semimono) [inline Assets/Scripts/Health.cs]: Hmmm... we could maybe add assists here.

File: Assets/Scripts/GUI/Menu.cs [hunk 4]
```diff
@@ -437,10 +442,12 @@ private enum state {
         //GetComponent<AudioListener>().enabled = false;
         //NetworkServer.SpawnObjects();
 	    GlobalConfig [] globalConfigs = Resources.FindObjectsOfTypeAll<GlobalConfig>();
-	    GlobalConfig globalConfig = globalConfigs[0];
+
+	    GlobalConfig globalConfig;
 	    if (globalConfigs.Length == 1) {
 		    globalConfig = Instantiate(globalConfigPrefab);
```
Comment (michbr) [inline Assets/Scripts/GUI/Menu.cs]: Resources.FindObjectsOfTypeAll<GlobalConfig>() includes prefabs. There's ALWAYS one. However, this is the only way to find de-activated gameobjects in the scene. So if there's ONE we create another. If there are two, we figure out which is the scene one and activate/spawn it.

**Pull request merged**